sdk/src/test/scala/com/horizen/SidechainMemoryPoolTest.scala

```diff
@@ -123,4 +156,59 @@ class SidechainMemoryPoolTestTestsExtension
       false)
     assertEquals("Size must be 2.", memoryPool.size, 2)
   }
+
+  @Test
+  def putWithMinFeeRate(): Unit = {
+
+    val list = List(
+      getRegularRandomTransaction(10,1),
+      getRegularRandomTransaction(20,2),
+      getRegularRandomTransaction(30,3),
+      getRegularRandomTransaction(40,1)
+    )
+    val minFeeRate  : Long = list.map(tx => new FeeRate(tx.fee(), tx.size()).getFeeRate()).min
+    val memoryPool = SidechainMemoryPool.createEmptyMempool(getMockedMempoolSettings(300, minFeeRate + 1))
+
+    assertEquals("Put tx1 operation must be failure.", false, memoryPool.put(list.apply(0)).isSuccess)
+    assertEquals("Put tx2 operation must be success.", true, memoryPool.put(list.apply(1)).isSuccess)
+    assertEquals("Put tx3 operation must be success.", true,  memoryPool.put(list.apply(2)).isSuccess)
+    assertEquals("Put tx4 operation must be success.", true, memoryPool.put(list.apply(3)).isSuccess)
+    assertEquals("MemoryPool must  have size 3 ", 3, memoryPool.size)
+  }
+
+  @Test
+  def putWithLimitedPoolSize(): Unit = {
+    //build a list of transactions up to 1MB size
+    var totalTxSizeMax = (1024*1024)  //1MB
+    val list = mutable.MutableList[RegularTransaction]()
+    val lowestFeeTx = getRegularRandomTransaction(10, 1)
+    list += lowestFeeTx
+    var totalTxSize = lowestFeeTx.size()
+    while (totalTxSize < totalTxSizeMax) {
+      var newTx = getRegularRandomTransaction(30, 1)
+      list += newTx
+      totalTxSize = totalTxSize + newTx.size()
+    }
+    val memoryPool = SidechainMemoryPool.createEmptyMempool(getMockedMempoolSettings(1, 0))
+
+    //put all transaction except the last one (so we are under 1MB limit)
+    for (i <- 0 to (list.size - 2)) {
+      assertEquals("Put tx operation must be success.", true, memoryPool.put(list.apply(i)).isSuccess)
+    }
+    assertEquals("MemoryPool must have correct size ", list.size-1, memoryPool.size)
+    assertEquals("Lowest fee transaction must be present ", true, memoryPool.getTransactionById(lowestFeeTx.id()).isPresent)
+
+    //add one more tx, causing the maxPoolSize to be reached and lowest one to be evicted
+    assertEquals("Put tx operation must be success.", true, memoryPool.put(list.apply(list.size - 1)).isSuccess)
+    assertEquals("MemoryPool must have correct size ", list.size-1, memoryPool.size)
+    assertEquals("Lowest fee transaction must not be present ", false, memoryPool.getTransactionById(lowestFeeTx.id()).isPresent)
+
+  }
+
```

Comment:
If you have time i would like to add a test case here where we add a tx with feeRate equals to the tx with the lowest feeRate already in the mempool.